### PR TITLE
Fix particles freezing by resetting particle timer

### LINKED ISSUE #56886: Vulkan: Manually emitted particles hide and freeze when the first one expires
### Godot version

v4.0.dev.20220105.official.8f6cba3a4 (have also tested in a unrelatedly modified 846c14eee )

### System information

Ubuntu 20.04

### Issue description

If one emits multiple particles through GDScript, all remaining particles will become invisible a small amount of time after the first particle of the series expires. If a new particle P is emitted, the invisible particles reappear at the exact same spot they hid and continue their trajectory, independent of time past. These particles will not become hidden, but particles emitted after P will hide/freeze again after P expires.

### Steps to reproduce

Create a slow-moving GPUParticle3D, and set up some manual emission trigger through GDScript, and emit multiple particles. Make sure you don't accidentally try to emit particles during the bug.

In the reproduction project, set the particle speed to 0.5 and press a directional key, to emit a particle, 8-9 times in quick succession and stop before the first particle disappears. You'll notice that all particles but the first will incorrectly disappear at around the same time. If you press a directional key again the particles that disappeared incorrectly will reappear for a short while before disappearing again, this time correctly.

### Minimal reproduction project

[Particle_emission.zip](https://github.com/godotengine/godot/files/7885618/Particle_emission.zip)

### PR DESCRIPTION
**Pull request by GanidhuAbey**
one line change, resets particle lifetime when new particle is emitted. closes #56886

### CODE REVIEW

File: servers/rendering/renderer_rd/renderer_storage_rd.cpp [hunk 1]
```diff
@@ -4825,6 +4825,7 @@ void RendererStorageRD::particles_request_process(RID p_particles) {
 
 	if (!particles->dirty) {
 		particles->dirty = true;
+        	particles->inactive_time = 0;
```
Comment (Calinou) [inline servers/rendering/renderer_rd/renderer_storage_rd.cpp:4828]: Indentation should use tabs, not spaces:

```suggestion
		particles->inactive_time = 0;
```

**Comment by Calinou** [comment]
Thanks for opening a pull request!

Please [amend the commit to use a human-readable commit message](https://github.com/godotengine/godot/blob/master/CONTRIBUTING.md#format-your-commit-messages-with-readability-in-mind) (`git commit --amend`). This way, people reading the Git log can easily figure out what's going on :slightly_smiling_face:

Once this is done, please squash the commits together so we can merge this. See [PR workflow](https://docs.godotengine.org/en/latest/community/contributing/pr_workflow.html#mastering-the-pr-workflow-the-rebase) for instructions :slightly_smiling_face:

**Comment by GanidhuAbey** [comment]
done, thank you

File: servers/rendering/renderer_rd/renderer_storage_rd.cpp [hunk 2]
```diff
@@ -4825,6 +4825,7 @@ void RendererStorageRD::particles_request_process(RID p_particles) {
 
 	if (!particles->dirty) {
 		particles->dirty = true;
+		particles->inactive_time = 0;
```
Comment (reduz) [inline servers/rendering/renderer_rd/renderer_storage_rd.cpp:4838]: This change will just make it impossible for particles to become inactive as long as they are visible, so definitely not a correct fix. The main issue is that I just dont understand the problem and cant see anything in the reproduction project of the original issue.

**Review by reduz — DISMISSED** [review]

File: servers/rendering/renderer_rd/renderer_storage_rd.cpp (hunk 2 above)
Comment (GanidhuAbey) [reply servers/rendering/renderer_rd/renderer_storage_rd.cpp:4838]: Hi,

Yea I definitely should have looked more into how that function was being called, I assumed that it was only being called when the a new particle was instanced and turns out, that's definitely wrong. The problem is, I am not seeing the behavior you're describing from this fix, the particle lasts only until its lifetime and disappears.

Any idea why this is happening?

**Review by clayjohn — CHANGES_REQUESTED** [review]
This appears to be the correct fix to me. 

I guess the issue was coming from a single frame where the particles would get set as inactive before this function made them active again in the following frame. 

It makes sense that `inactive_time`` is reset after each emission as that particle should be allowed to stay active for its lifetime.

Before merging, you will need to delete the print statement you added and squash your commits together. The whole thing should contain just a single commit (for instructions on how to do so, see here: https://docs.godotengine.org/en/latest/community/contributing/pr_workflow.html#the-interactive-rebase)

**Review by clayjohn — APPROVED** [review]
Thanks! Everything is looking good now

**Comment by GanidhuAbey** [comment]
Cool, it took a lot of time to learn how to use git properly, haha.

It's exactly as you said, the system was getting set to inactive too early, I figured that the best way to deal with this was resetting the inactive_time and letting the new particle survive it's entire lifetime.

Thanks for all your help in this, had a huge breakthrough with the problem while talking with you about it :)

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**